### PR TITLE
process _re2.cc and add tox driver and github workflow ci files

### PR DESCRIPTION
**Pull request by sarnold**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -61,8 +64,13 @@ def include_dirs():
...
+    ext_modules=ext_modules,
+    # Currently, build_ext only provides an optional "highest supported C++
+    # level" feature, but in the future it may provide more features.
+    cmdclass={"build_ext": build_ext},
+    zip_safe=False,
     py_modules=['re2'],
-    python_requires='~=3.6',
-    install_requires=['six'],
+    python_requires='>=3.6',
+    install_requires=['pybind11', 'six'],
```
Comment (henryiii) [inline setup.py:74]: Unless something is really unusual, pybind11 is only needed when you build (so put it In pyproject.toml), not when you install.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      run: |
+        brew install re2
+      if: startsWith(matrix.os, 'mac')
+    - name: Install PPA deps
+      run: |
+        sudo apt-get -qq update
+        sudo apt-get install -y software-properties-common
+        sudo add-apt-repository -y -s ppa:nerdboy/embedded
+        sudo apt-get install -y libre2-dev
+      if: startsWith(matrix.os, 'ubuntu')
```
Comment (henryiii) [inline .github/workflows/ci.yml:64]: I'd use `runner.os == 'Linux'` here, it's general.

**Comment by henryiii** [comment]
You might want to try cibuildwheel, see https://github.com/pybind/python_example/blob/master/.github/workflows/wheels.yml - you can use env vars to run pre-setup commands such as those downloading dependencies.

File: setup.py (hunk 1 above)
Comment (sarnold) [reply setup.py:87]: I think I added that when I was poking at windows, thanks for the reminder ;)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sarnold) [reply .github/workflows/ci.yml:65]: I'm not sure I'll ever get to the docker/manylinux stuff but cibuildwheel looks interesting.  Thanks for the feedback.

**Pull request merged**